src/components/navbar.js

```diff
@@ -18,11 +18,45 @@ const activeNavIcon = (path) => ({
 });
 
 function Navbar({pages, darkMode, setDarkMode}) {
+  if (window.innerWidth < 769) {
+    let prevScrollpos = window.pageYOffset;
+    window.onscroll = function () {
```

Comment:
I'm interested to know the implications of this scroll even handler, would it function only within the navbar component, and would it lead to any side effects in the scrolling of the home page (like scrolls getting cancelled and not being able to scroll smoothly)

Reply:
Hi, @jeremyphilemon 
this scroll event will apply in window width less than 769px and it will apply to all windows wherever navbar component is there.
``` javascript
if (window.innerWidth < 769) {
 ....    // this part of code in navbar can be moved to some other components also, but i found it suitable inside navbar component.
}
```

**regarding side effects** in the scrolling of the pages, i don't think it's going to effect. i have tested in this 7 browsers it's working smoothly as it should work.
![WhatsApp Image 2020-04-24 at 11 09 42 PM](https://user-images.githubusercontent.com/45959932/80241856-fb1b6080-8681-11ea-8186-55d7ba01614d.jpeg)

Reply:
```
if (window.innerWidth < 769) {
}
```
This comparison fires for scroll of every pixel – even if the value is greater than 769, it still fires.

How about we use intersectionObserver API to detect if the user is away from the top part of the screen. and then activate the sticky?

Resource: https://github.com/streamich/react-use/blob/master/docs/useMeasure.md

Reply:
> i have changed few things in code to improve performance of scroll event.
* i moved `scroll event and if condition` code from navbar.js to App.js because in navbar `if condition` code was triggered each time when navbar get opened and when route changes.
* used setInterval to manipulate DOM inspite of manipulating DOM continously when scroll.

@jeremyphilemon 
![Screenshot from 2020-04-25 12-16-27](https://user-images.githubusercontent.com/45959932/80273298-a661ff00-86ee-11ea-945e-750f353e7a72.png)
> i tried logging the things to see when if condition and code inside setInterval is getting triggered.

i found that `if condition code` is running only once when App get mounted even for value greater than 769px and on resizing.

![scroll](https://user-images.githubusercontent.com/45959932/80273762-b1b72980-86f2-11ea-828d-66402ce14171.gif)

i can use the resources you shared, but i don't think it's going to make any big difference.
**Further your any advice and suggestions will be greatly appreciated.**